Defining component for internacionalization

breakingstones opened the pull request:
Internationalization component:
'react-language-kit'
https://github.com/znti/react-language-kit

**component demo in action on the "Tables" page**

This component makes use of 'intl-messageformat' lib. For more details see:
https://github.com/formatjs/intl-messageformat
This lib is based on the specification ECMAScript Internationalization API 4.0
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl

Features in 'react-language-kit':
- Display numbers, currency, dates and times for different locales.
- Pluralize labels in strings.
- Support variables in message.
- Message format is strictly implemented by ICU standards.
- Locale data in nested JSON format are supported.
- fallback to default language if it doesn't find the term you're looking for

Included support for Class Components (does not use hooks)
Uses context API where Consumers are used as HOCs for the component to internationalize

Included support for use with Vanilla Javascript
(This function does not use hooks):
 - useTranslationTo
Note: Its necessary to enter the locale manually.

breakingstones committed:
Defining component for internacionalization

Internationalization component:
'react-language-kit'

This component makes use of 'intl-messageformat' lib. For more details see:
https://github.com/formatjs/intl-messageformat
This lib is based on the specification ECMAScript Internationalization API 4.0
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl

Features in 'react-language-kit':
- Display numbers, currency, dates and times for different locales.
- Pluralize labels in strings.
- Support variables in message.
- Message format is strictly implemented by ICU standards.
- Locale data in nested JSON format are supported.
- fallback to default language if it doesn't find the term you're looking for

Included support for Class Components (does not use hooks)
Uses context API where Consumers are used as HOCs for the component to internationalize

Included support for use with Vanilla Javascript
(This function does not use hooks):
 - useTranslationTo
Note: Its necessary to enter the locale manually.

znti commented on src/Components/TopBar/_styles.js:
```diff
@@ -18,6 +18,15 @@ const styles = (theme) => ({
+        paddingBottom: theme.spacing(1),
+        paddingLeft: theme.spacing(2),
+        paddingRight: theme.spacing(2),
+        height: 'unset',
+        background: 'unset',
+        boxShadow: 'unset',
```
Why would you need to explicitly set those?

breakingstones replied:
There are a set of custom styles that are applied on the Material UI Theme:
[/src/theme.js](https://github.com/cfrancisco/gui-builder/blob/dev/src/theme.js#L29)
In the specific case of the buttons, there is the following overrides:
`
MuiButton: {
    root: {
        background: 'linear-gradient(45deg, #6b91fe 30%, #53c7ff 90%)',
        height: 48,
        boxShadow: '0 3px 5px 2px rgba(105, 137, 255, 0.3)',
    },
},
`
It might be interesting to review these style overrides on 'MuiButton.root'.

breakingstones committed:
Commit to code quality improvement

Treatment to reduce line length below 100 characters
to comply with CodeFactor and Codacy rules.

nunesjoab commented on src/Components/TopBar/TopBar.js:
```diff
@@ -52,6 +96,42 @@ const TopBar = ({ handleDrawerToggle, open }) => {
                 {auth && (
                     <>
+                        <Button
+                            aria-controls="customized-menu"
```
This aria-controls property must have as value the ID of the Menu component. There is not?

breakingstones replied:
Done

nunesjoab commented on src/Components/TopBar/_styles.js:
```diff
@@ -18,6 +18,15 @@ const styles = (theme) => ({
+        paddingTop: theme.spacing(1),
+        paddingBottom: theme.spacing(1),
+        paddingLeft: theme.spacing(2),
+        paddingRight: theme.spacing(2),
```
Here, you can use shorthand form to define padding attribute: `padding: ``${theme.spacing(1)}px ${theme.spacing(2)}px``,`

breakingstones replied:
done like this:
`padding: theme.spacing(1, 2)`
according [Material-ui](https://material-ui.com/pt/customization/spacing/#aridade-m%C3%BAltipla)

nunesjoab commented on src/Pages/Tables/Tables.js:
```diff
@@ -10,15 +12,27 @@ const useStyles = makeStyles(styles);
 
 const Tables = () => {
     const classes = useStyles();
+    const t = useTranslation(i18nMap);
```
Try using meaningfully named variables. Letters say nothing about what these variables are or do.

breakingstones replied:
done

nunesjoab commented on src/Pages/Tables/i18n.js:
```diff
@@ -0,0 +1,168 @@
+export default {
```
The name of this file could be different, like locale.

breakingstones replied:
I defined as 'locales'

nunesjoab replied:
Good!

breakingstones committed:
Improvements according to comments

Improvements applied as suggested.

code-factor committed:
[CodeFactor] Apply fixes to commit 3e1123b
